common/src/main/java/com/wynntils/functions/HorseFunctions.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.functions;
+
+import com.wynntils.core.functions.ActiveFunction;
+import com.wynntils.core.managers.Model;
+import com.wynntils.mc.utils.InventoryUtils;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wc.custom.item.ItemStackTransformModel;
+import com.wynntils.wc.custom.item.WynnItemStack;
+import com.wynntils.wc.custom.item.properties.ItemProperty;
+import java.util.List;
+import net.minecraft.world.item.ItemStack;
+
+public class HorseFunctions {
+
+    private static WynnItemStack getHorse() {
+        int horseSlot = InventoryUtils.findHorseSlotNum();
+        if (horseSlot == -1) return null;
+        ItemStack is = McUtils.player().getInventory().getItem(horseSlot);
+        if (!(is instanceof WynnItemStack horse)) return null;
+        return horse;
+    }
+
+    public static class HorseLevelFunction extends ActiveFunction<String> {
+        @Override
+        public String getValue(String argument) {
+            WynnItemStack horse = getHorse();
+            if (horse == null) return "??";
+            return String.valueOf(horse.getProperty(ItemProperty.HORSE).getLevel());
+        }
+
+        @Override
+        public List<Class<? extends Model>> getModelDependencies() {
+            return List.of(ItemStackTransformModel.class);
+        }
+
+        @Override
+        public List<String> getAliases() {
+            return List.of("h_lvl");
+        }
+    }
+
+    public static class HorseLevelMaxFunction extends ActiveFunction<String> {
```

Comment:
```suggestion
    public static class HorseLevelMaxFunction extends ActiveFunction<Integer> {
```
and then you can skip the String.valueOf.

Same goes for all the other horse functions as well, except `HorseNameFunction`.